src/command-base.ts

```diff
@@ -1,20 +1,19 @@
-import Command from '@oclif/command'
-import {createEnv} from 'yeoman-environment'
+/*
+ * Copyright (c) 2022, salesforce.com, inc.
+ * All rights reserved.
+ * Licensed under the BSD 3-Clause license.
+ * For full license text, see LICENSE.txt file in the repo root or https://opensource.org/licenses/BSD-3-Clause
+ */
+
+import { Command } from '@oclif/core';
+import { createEnv } from 'yeoman-environment';
 
 export default abstract class CommandBase extends Command {
-  protected async generate(type: string, generatorOptions = {}) {
-    const env = createEnv()
+  protected async generate(type: string, generatorOptions = {}): Promise<void> {
+    const env = createEnv();
 
-    env.register(
-      require.resolve(`./generators/${type}`),
-      `sfdx:${type}`,
-    )
+    env.register(require.resolve(`./generators/${type}`), `sfdx:${type}`);
 
-    await new Promise((resolve, reject) => {
-      env.run(`sfdx:${type}`, generatorOptions, (err: Error, results: any) => {
-        if (err) reject(err)
-        else resolve(results)
-      })
-    })
+    await env.run(`sfdx:${type}`, generatorOptions);
```

Comment:
`yeoman/environment@3` migrated to promises:
https://github.com/yeoman/environment/releases/tag/v3.0.0